include/envoy/secret/secret_manager.h

```diff
@@ -17,14 +21,37 @@ class SecretManager {
   virtual ~SecretManager() {}
 
   /**
+   * @param secret a raw pointer of the derived class of Secret.
+   */
+  template <typename T> void addOrUpdateSecret(T* secret) {
```

Comment:
Using raw pointer in the function argument is NOT so clear about the ownership transfer.  I think this function not need to be template function,  it can be a pure virtual function to pass in SecretSharedPtr.